src/mappings/serviceAgreement.ts

```diff
@@ -30,7 +32,7 @@ export async function handleServiceAgreementCreated(
   endTime.setSeconds(endTime.getSeconds() + period.toNumber());
 
   const sa = ServiceAgreement.create({
-    id: event.args.serviceAgreement,
+    id: eventServiceAgreementId,
```

Comment:
This is the updated agreement structure, add the `planId` and `planTemplateId` in the entity as well

```
struct ClosedServiceAgreementInfo {
    address consumer;
    address indexer;
    bytes32 deploymentId;
    uint256 lockedAmount;
    uint256 startDate;
    uint256 period;
    uint256 planId;
    uint256 planTemplateId;
}
```

Reply:
updated